### PR TITLE
GKDTrainer: Fix return_outputs in Liger kernel path and update tests

### PR DESCRIPTION
**Pull request by roycho96**
# What does this PR do?

Fixes `UnboundLocalError` when `return_outputs=True` in GKDTrainer's Liger kernel path, and updates tests for Liger kernel compatibility.

## Changes

1. **Bug fix**: Handle `return_outputs=True` in Liger kernel path
   - `student_outputs` was deleted before being returned in `compute_loss`
   - Now properly performs forward pass when outputs are needed

2. **Test update**: Remove `xfail` marker from `test_gkd_trainer_with_liger`
   - OOM issue doesn't happen in liger-kernel 0.6.4

3. **New test**: Add `test_compute_loss_return_outputs_with_liger`
   - Ensures `evaluate()` works correctly with Liger kernel

Tested with liger-kernel 0.6.4. All GKD tests pass.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Medium Risk**
> Touches `GKDTrainer.compute_loss` in the Liger fused-loss path, which is core training/eval logic; incorrect outputs could affect evaluation behavior, but the change is small and covered by targeted tests.
> 
> **Overview**
> Fixes `GKDTrainer.compute_loss` when using the Liger fused JSD loss so `return_outputs=True` no longer fails (keeps `student_outputs` alive, returns a `ModelOutput`, and frees memory/empties cache after loss computation).
> 
> Updates Liger-related tests by removing the previous `xfail` on the Liger training test and adding a new regression test asserting `evaluate()` works on the Liger path (i.e., `return_outputs=True` does not raise and produces `eval_loss`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 36e2fc9ccf488eacde3869c8a2a21c8d0580d60a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by kashif — APPROVED** [review]

File: tests/experimental/test_gkd_trainer.py [hunk 1]
```diff
@@ -279,3 +278,37 @@ def test_generation_config_init(self):
...
+            processing_class=self.tokenizer,
+        )
+
+        if not getattr(trainer, "use_liger_gkd_loss", False):
+            pytest.skip("Liger fused JSD not enabled at runtime")
+
+        # evaluate() calls compute_loss with return_outputs=True
+        # This should not raise UnboundLocalError
+        eval_results = trainer.evaluate()
+
```
Comment (kashif) [inline tests/experimental/test_gkd_trainer.py]:
```suggestion

```

**Comment by roycho96** [comment]
@kashif 
Gentle ping on this. Let me know if there’s anything I can help with to get it over the line.

**Comment by kashif** [comment]
@codex review

**Comment by kashif** [comment]
I was thinking perhaps a simpler fix could be to store the logits before deleting them  in the liger path, and then use them for returning the outputs instead of re-runing the model() forward?

**Comment by roycho96** [comment]
> I was thinking perhaps a simpler fix could be to store the logits before deleting them in the liger path, and then use them for returning the outputs instead of re-runing the model() forward?

Thanks for the suggestion! Applied it in the latest commit.

File: tests/experimental/test_gkd_trainer.py [hunk 2]
```diff
@@ -280,3 +279,36 @@ def test_generation_config_init(self):
...
+        trainer = GKDTrainer(
+            model=self.model_id,
+            teacher_model=self.model_id,
+            args=training_args,
+            train_dataset=dummy_dataset["train"],
+            eval_dataset=dummy_dataset["test"],
+            processing_class=self.tokenizer,
+        )
+
+        if not getattr(trainer, "use_liger_gkd_loss", False):
```
[the remark this answers is not in the document]
Comment (roycho96) [reply tests/experimental/test_gkd_trainer.py]: Done

File: trl/experimental/gkd/gkd_trainer.py [hunk 3]
```diff
@@ -355,6 +355,17 @@ def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=N
 
             # Release hidden states after loss computation
             del student_hidden, teacher_hidden, true_labels
+
+            if return_outputs:
+                student_outputs = model(
+                    input_ids=inputs["input_ids"],
+                    attention_mask=inputs["attention_mask"],
+                )
```
[the remark this answers is not in the document]
Comment (roycho96) [reply trl/experimental/gkd/gkd_trainer.py]: Removed

**Review by kashif — APPROVED** [review]
